### DISCUSSION TITLE
using SSH vs. https on github to connect repo

### DISCUSSION THREAD

**Discussion by tsekeres** [nss-evening-cohort-14/evening-client-side#30] [Help Me!] [answered]
Tad Sekeres
Have you searched the other help tickets? Yes
Describe your issue:
1. When I attempted to test my code with the (npm test main.test.js) command, I got an error message.
2. I am worried that I did not select SSH when copying the dir title. Just carry on and see if it connects?

List any error messages that appear:
➜  exercises cd js-practice-1-tsekeres 
➜  js-practice-1-tsekeres git:(attempt1) npm test main.test.js

> example@0.0.0 test /mnt/c/Users/Owner/workspace/foundations/exercises/js-practice-1-tsekeres
> jest "main.test.js"

sh: 1: jest: not found
npm ERR! Test failed.  See above for more details.
➜  js-practice-1-tsekeres git:(attempt1) 

What have you googled?  Didn't think this was googlable. Thought this might be specific to this process.

What have you tried? Checked to see if I mistyped the command. Nothing after that.

Copy/paste link TO THE FILE on your branch or in your repo: this is in my VS Code terminal.

**Comment by drteresavasquez** [comment]
Did you follow all the steps on the readme file? Specifically the `npm install` command?

**Reply by tsekeres** [reply]
Yes. But regarding my #2 question, selecting the SSH tab before copying the text in the window was not on the instructions and I don't think I did that.

**Reply by drteresavasquez** [reply]
So were you able to complete this? The clone command in the readme was a part of grabbing the ssh key, which was something I walked everyone through in class and one of the reasons I wanted all to watch us go through the setup. If you still need help, let us know.

**Reply by tsekeres** [reply]
i'm currently reading stackoverflow and trying to reinstall npm. I appreciate the walk through during class. I was stuck and in "listening mode" at the clone moment. And then later, without the video to go with yet, I just missed that detail. Should I redo the whole process to include the SSH selection somehow?

**Reply by drteresavasquez** [reply]
You should not have to reinstall npm. The video is up now, so maybe walk through it. You can delete your local directory and reconnect again. The github repo and everything should be fine, so just start from the beginning of the video (9 minutes long) and work through it AFTER you delete your local `js-practice-1` directory

**Reply by tsekeres** [reply]
I do need help. I'm not even sure how to do some of the suggestions in the stack answers.

I have:
-typed "npm install" and got lots of errors. I did this in my branch, main and in exercises.
-typed "npm list -g --depth=0" to check if npm is installed globally. I saw "── npm@6.14.4" in the /usr/lib
-typed "npm list --depth=0" to check the same locally and got lots of errors that mention node_modules
-typed "npm --save install jest" and got "npm ERR! Maximum call stack size exceeded"

this is also suggested "If local, you might need to provide the absolute path node_modules/bin/jest". But I don't know hwo to do that.

**Reply by drteresavasquez** [reply]
if you are getting errors on npm installs...what are you supposed to try first? it rhymes with summo...

**Reply by drteresavasquez** [reply]
And please follow the README on your repo.

**Reply by tsekeres** [reply]
So just literally try what you tried again with sudo as a prefix whenever you get an error?

**Reply by drteresavasquez** [reply]
Yep. If the error includes anything about access privileges on a npm install always try sudo first. After that, you'll know if you have another issue.

**Reply by tsekeres** [reply]
Ok. I'm sorry. But I am back at the same spot after deleting local directory and reconnecting. 

I started with the clone process. I tried again to install npm. I got permissions error messages and tried sudo npm install. After using sudo, I got:
npm ERR! Maximum call stack size exceeded

So I looked that up. After reading several answers. I tried: npm cache clean. Then after the expected error tried: npm cache clean --force

That told me: npm WARN using --force I sure hope you know what you are doing.

Which I definitely don't. So then I tried to install npm again. And got the same: "npm ERR! Maximum call stack size exceeded" .

So I remembered that you said I shouldn't need to reinstall npm. and I just went ahead with a test. And I got the same exact message:


➜  js-practice-1-tsekeres git:(attempt1) npm test main.test.js

> example@0.0.0 test /mnt/c/Users/Owner/workspace/foundations/exercises/js-practice-1-tsekeres
> jest "main.test.js"

sh: 1: jest: not found
npm ERR! Test failed.  See above for more details.got to npm install and tried again to install. Got

**Reply by drteresavasquez** [reply]
You are inside the js-practice-1 directory as you're running your commands right? If so, we may need to look at this on Tuesday.

**Comment by drteresavasquez** [comment]
Also, any time your terminal gives you an error, it’s is googleable. Always try to google first. This is the first thing that pops up when I search: `sh: 1: jest: not found
npm ERR! Test failed. See above for more details.`

https://stackoverflow.com/questions/46918027/npm-test-jest-not-working

**Reply by tsekeres** [reply]
Understood. My B.

**Comment by drteresavasquez** [comment]
And then this: https://stackoverflow.com/questions/56281158/npm-err-test-failed-not-able-to-run-test

**Comment by JuanDavila1101** [comment]
I just added a new SSH key from Ubuntu.
Ubuntu was not seeing the directories for ssh in the .ssh directory 
I created a new key and added it to the GitHub SSH keys 
and updated the Git Config permission with the sudo from Ubuntu, now my SSH work and I don't need sudo for the local server
I used the links from the course in Git Bash, did it again on Ubuntu for it to work there as well 
https://www.youtube.com/watch?v=znRMcNG9_qQ&feature=youtu.be
https://www.youtube.com/watch?v=8hlmIObpMd4&feature=youtu.be
https://www.youtube.com/watch?v=exva3J_jojc&feature=youtu.be

**Comment by drteresavasquez** [comment]
@tsekeres watch this video to help you complete the assignment without needing the code on your local machine. It is a little more cumbersome, but at least it gets you an option to complete the assignment. We can discuss your issue and take a look on Tuesday.

https://vimeo.com/501565182/59d6c13c20

**Comment by drteresavasquez** [comment]
@tsekeres it looks like you were able to resolve your issue. Can you let us know what you did to get it completed? Thanks!

**Comment by tsekeres** [comment]
I truly do not know what fixed this issue. I thought it had not been fixed at all up until trying it YET AGAIN today. And for some reason it works now. I tried many, many things.

After reading the suggestions in Sara Schoonover's ticket I last tried deleting  the node_modules directory and package.lock.JSON file. Then again reinstalled the npm dealios. It did not work and I used the work around directly in Github from the video Dr. T put out to turn in my work. Went to bed, then got a message from Dr T that said it looked like it was fixed. I don't know why she thought that. But after restarting computer and sitting overnight I guess something finally took.

**Reply by nashvegasnate** [reply]
Tad: I was having issues with the local not recognizing any of the answers as correct that I knew for a fact were correct. It kept running and telling me that they were all incorrect. I deleted the node_modules directory and the package.lock.JSON file as well, then did npm install, but it didn't matter. I found another potential fix by reinstalling homebrew, but now I'm getting even more errors. All of them titled "npm ERR!" with different messages attached. Now feeling like I did something I shouldn't have.

**Reply by drteresavasquez** [reply]
What are the errors? Can you paste them here?

**Reply by drteresavasquez** [reply]
@nashvegasnate in looking at your `attempt1` branch, you have some things that are out of order and extra line breaks. For example, you have the return, but then some other text that is not commented out below it. You only need the answer to the problem otherwise you may get errors. Here is some of what you have:

**You don't need the `returnString` item here.** 

```
// 7
function returnString(str) {
  return(str)
  //simply return the string provided: str
  returnString
}
```

**You have extra spacing and line breaks and then the random `add` at the end of this.**

```
// 8
function add(x, y) {
  return x + y;
 add(5, 7);
  // x and y are numbers
  // add x and y together and return the value
  // code here

 add

}
```

**Comment by Gabrielle-Tobermann** [comment, marked as answer]
If you get the error "git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository." 

Use the HTTPS link (Github) instead of SSH link. In terminal type: 

git remote set-url origin [https link from Github]

**Comment by nashvegasnate** [comment]
*I figured it out! I was missing a "}" in question #12. Now I can finally
get on with my life!*
